server/service/s3_object_presigner/s3_object_presigner.go

```diff
@@ -67,15 +68,21 @@ func (p *S3ObjectPresigner) CheckAndPresign(ctx context.Context) {
 			continue
 		}
 
-		presignURL, err := s3Client.PresignGetObject(ctx, s3ObjectPayload.Key)
-		if err != nil {
-			return
+		var referenceURL string
```

Comment:
Could we skip this? This is the scheduler that refreshes the presigned time. For resources that use custom domain, we don't need to reference its URL.

Reply:
I kept this because when I set/unset custom domains, this scheduler will automatically update and switch to using the custom domain URL/presigned URL after the previous signature expires.